testing/pom.xml

```diff
@@ -220,12 +222,81 @@
           <forkCount>1</forkCount>
           <systemPropertyVariables>
             <stargate.libdir>${basedir}/../stargate-lib</stargate.libdir>
+            <jacoco-agent.destfile>${project.build.directory}/jacoco-it.exec</jacoco-agent.destfile>
+            <jacoco-agent.args>${failsafe.jacoco.args}</jacoco-agent.args>
           </systemPropertyVariables>
+          <argLine>${failsafe.jacoco.args}</argLine>
           <includes>
             <include>**/*.java</include>
           </includes>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>report-aggregate</id>
+            <phase>verify</phase>
+            <goals>
+              <goal>report-aggregate</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>merge-results</id>
+            <phase>verify</phase>
+            <goals>
+              <goal>merge</goal>
+            </goals>
+            <configuration>
+              <fileSets>
+                <fileSet>
+                  <directory>${code.coverage.project.folder}</directory>
+                  <includes>
+                    <include>**/target/*.exec</include>
+                  </includes>
+                </fileSet>
+              </fileSets>
+              <destFile>${code.coverage.overall.data.folder}/aggregate.exec</destFile>
+            </configuration>
+          </execution>
+          <execution>
+            <id>post-merge-report</id>
+            <phase>verify</phase>
+            <goals>
+              <goal>report</goal>
+            </goals>
+            <configuration>
+              <includes>
+                <include></include>
+              </includes>
+              <dataFile>${code.coverage.overall.data.folder}/aggregate.exec</dataFile>
+              <outputDirectory>${code.coverage.overall.data.folder}/jacoco</outputDirectory>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <!-- Unpack all classes for all stargate dependencies, otherwise jacoco ignores classes it cant find in this module -->
```

Comment:
Is it possible to explicitly tell jacoco the name of our package and have it instrument all classes matching the package name? That would be cleaner that unpacking, I guess... I hope it's possible, but I'm not sure :thinking: